server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1284,53 +1277,53 @@ private static IndexMetadata.State excludeState(IndicesOptions options) {
             if (Regex.isMatchAllPattern(expression)) {
-                return filterIndicesLookup(context, metadata.getIndicesLookup(), null, context.getOptions());
+                return filterIndicesLookup(indicesLookup, null, context.getOptions().ignoreAliases(), context.includeDataStreams());
             } else if (expression.indexOf("*") == expression.length() - 1) {
-                return suffixWildcard(context, metadata, expression);
+                return suffixWildcard(context, indicesLookup, expression);
             } else {
-                return otherWildcard(context, metadata, expression);
+                return filterIndicesLookup(
+                    indicesLookup,
+                    e -> Regex.simpleMatch(expression, e.getKey()),
+                    context.getOptions().ignoreAliases(),
+                    context.includeDataStreams()
+                );
             }
         }
 
-        private static Map<String, IndexAbstraction> suffixWildcard(Context context, Metadata metadata, String expression) {
+        private static Map<String, IndexAbstraction> suffixWildcard(
+            Context context,
+            SortedMap<String, IndexAbstraction> indicesLookup,
+            String expression
+        ) {
             assert expression.length() >= 2 : "expression [" + expression + "] should have at least a length of 2";
             String fromPrefix = expression.substring(0, expression.length() - 1);
             char[] toPrefixCharArr = fromPrefix.toCharArray();
             toPrefixCharArr[toPrefixCharArr.length - 1]++;
             String toPrefix = new String(toPrefixCharArr);
-            SortedMap<String, IndexAbstraction> subMap = metadata.getIndicesLookup().subMap(fromPrefix, toPrefix);
-            return filterIndicesLookup(context, subMap, null, context.getOptions());
-        }
-
-        private static Map<String, IndexAbstraction> otherWildcard(Context context, Metadata metadata, String expression) {
-            final String pattern = expression;
-            return filterIndicesLookup(
-                context,
-                metadata.getIndicesLookup(),
-                e -> Regex.simpleMatch(pattern, e.getKey()),
-                context.getOptions()
-            );
+            SortedMap<String, IndexAbstraction> subMap = indicesLookup.subMap(fromPrefix, toPrefix);
+            return filterIndicesLookup(subMap, null, context.getOptions().ignoreAliases(), context.includeDataStreams());
         }
 
         private static Map<String, IndexAbstraction> filterIndicesLookup(
-            Context context,
-            SortedMap<String, IndexAbstraction> indicesLookup,
+            Map<String, IndexAbstraction> indicesLookup,
             Predicate<? super Map.Entry<String, IndexAbstraction>> filter,
-            IndicesOptions options
+            boolean ignoreAliases,
+            boolean includeDataStreams
```

Comment:
minor: why not keep the Context here and pull the information (indicesLookup, ignoreAliases, includeDataStreams) from that ? It would be more inline with the other refactoring here.

Reply:
The `filterIndicesLookup` method doesn't always act on the indices lookup from the context, see `suffixWildcard`.
I have pushed https://github.com/elastic/elasticsearch/pull/89000/commits/[781a768](https://github.com/elastic/elasticsearch/pull/89000/commits/781a7685688842c0de9c5efb867016dcec14da41)5688842c0de9c5efb867016dcec14da41 to make suffixWildcard not use the context.